mlkem/rej_uniform.c

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "params.h"
+
+#include "asm/asm.h"
+#include "rej_uniform.h"
+
+/*************************************************
+* Name:        rej_uniform_scalar
+*
+* Description: Run rejection sampling on uniform random bytes to generate
+*              uniform random integers mod q
+*
+* Arguments:   - int16_t *r:          pointer to output buffer
+*              - unsigned int len:    requested number of 16-bit integers
+*                                     (uniform mod q)
+*              - const uint8_t *buf:  pointer to input buffer
+*                                     (assumed to be uniform random bytes)
+*              - unsigned int buflen: length of input buffer in bytes
+*
+* Returns number of sampled 16-bit integers (at most len)
+**************************************************/
+static unsigned int rej_uniform_scalar(int16_t *r,
```

Comment:
Do you think it makes sense to keep this C code closer to what the assembly is doing? That would make it easier to follow what's happening in the asm.

Reply:
I had the intrinsic version, however, I think we should keep the scalar version as-is, and either ASM or instrinsic, preferably ASM version.

Reply:
I agree, we want to stay close to the reference implementation.